src/ngraph/ngraph_emitter.cc

```diff
@@ -695,6 +695,42 @@ void Emitter::CreateBinaryOps() {
     return cast_result(CreateAutoBroadcast<ngraph::op::LessEq>(node),
                        getType(node->dtype_));
   };
+  ngraph_op_funcs_["smooth_l1"] = [this](const NodePtr& node) {
+    /* Smooth L1 Loss is a loss specific for R-CNN franchise training
+     * Smooth L1 Loss function:
+     * f(x) = 0.5 * (sigma * x) ^ 2,     |x| < 1 / sigma^2
+     *      = |x| - 0.5 / sigma / sigma, otherwise
+     * When sigma = 1, it is equivalent to the Huber loss, evaluated at
+     * delta = 1.
+     * smooth_l1_loss = w_out * f(w_in * x)
+     * with w_in, w_out provided by input_data.
+     */
+    auto input = op_map_[node->inputs_[0]];
+    auto sigma =
+        makeConstant(node, get_default(node, "scalar", std::string("0")));
+    auto sigma_sq = sigma * sigma;
+    auto inv_sigma_sq = std::make_shared<ngraph::op::Divide>(
+        makeConstant(node, "1.0"), sigma_sq);
+
+    // check if input is greater than inv_sigma_sq
+    auto is_input_gt_sigma =
+        std::make_shared<ngraph::op::Greater>(input, inv_sigma_sq);
+
+    // f(x) = 0.5 * (sigma * x) ^ 2
+    auto result_input_sq = makeConstant(node, "0.5") * input * input * sigma_sq;
+    // f(x) = x - 0.5 / sigma / sigma
+    auto result_input_gt = input - makeConstant(node, "0.5") * inv_sigma_sq;
+    // f(x) = -x - 0.5 / sigma / sigma
+    auto result_input_lt = -input - makeConstant(node, "0.5") * inv_sigma_sq;
```

Comment:
Would it make sense to define just one nGraph node to represent the constant 0.5?  I don't know either way in terms of how that impacts nGraph's performance at graph-compile-time and graph-execution-time.

Reply:
yep, will refactor that.